File: Android-MVP-Java/app/src/main/java/com/base/mvp/core/base/BaseViewHolder.java
package com.base.mvp.core.base;

import android.view.View;
import androidx.recyclerview.widget.RecyclerView;

public abstract class BaseViewHolder extends RecyclerView.ViewHolder {

    private int currentPosition;

    public BaseViewHolder(View itemView) {
        super(itemView);
    }

    protected abstract void clear();

    public void onBind(int position) {
        currentPosition = position;
        clear();
    }

    public int getCurrentPosition() {
        return currentPosition;
    }
}

File: Android-MVVM-Kotlin/app/src/main/java/com/base/mvvm/core/adapter/BaseViewHolder.kt
package com.base.mvvm.core.adapter

import android.view.View

import androidx.recyclerview.widget.RecyclerView

abstract class BaseViewHolder(itemView: View?) : RecyclerView.ViewHolder(itemView!!) {

    var currentPosition = 0
        private set

    abstract fun clear()

    open fun onBind(position: Int) {
        currentPosition = position
        clear()
    }

}